app/src/main/java/tech/relaycorp/letro/onboarding/registration/RegistrationDomainProvider.kt

```diff
@@ -5,21 +5,34 @@ import javax.inject.Inject
 
 interface RegistrationDomainProvider {
     fun getDomain(): String
+    fun getDomainLocale(): Locale
 }
 
 class RegistrationDomainProviderImpl @Inject constructor() : RegistrationDomainProvider {
 
+    private val lazyDomainLocale: Locale by lazy {
+        Locale.getDefault()
+    }
+
     private val lazyDomain: String by lazy {
         val locale = Locale.getDefault()
-        when (locale.toString()) {
-            "en_GB" -> "@cuppa.fans"
-            "en_US" -> "@applepie.rocks"
-            "es_VE" -> "@guarapo.cafe"
-            else -> "@nautilus.ink"
-        }
+        DOMAIN_BY_LOCALE[locale.toString()] ?: FALLBACK_DOMAIN
```

Comment:
Let's get the locale from `lazyDomainLocale`, so this class would return consistent values to the callers

```
    private val lazyDomain: String by lazy {
        DOMAIN_BY_LOCALE[lazyDomainLocale.toString()] ?: FALLBACK_DOMAIN
    }
```